docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+
+- When the user is _admin_ and specified `add_all_backend_roles` to true, there’ll be an exception: `Admin users cannot add all backend roles to a model group.`
+
+- When a user doesn’t have any backend roles and specified `add_all_backend_roles to true` and model_access_mode to _restricted_, there’ll be an exception: `You must have at least one backend role to register a restricted model group.`
+
+- When a user doesn’t specify either of `add_all_backend_roles` or `backend_roles` fields and set model_access_mode to _restricted_, there’ll be an exception: `You must specify one or more backend roles or add all backend roles to register a restricted model group.`
+
+- When a user specified both `add_all_backend_roles` and `backend_roles` and set model_access_mode to _restricted_, there’ll be an exception: `You cannot specify backend roles and add all backend roles at the same time`
+
+- When a user specified `backend_roles` and set model_access_mode to _restricted_, but the user doesn’t have all the backend_roles specified, there’ll be an exception: `You cannot specify backend roles and add all backend roles at the same time`
+
+### Note
+- If security or `model_access_control_enabled` setting is disabled in the cluster, every user is null and by default, in such cluster, all model groups are public. User can register a model group with a `name` and `description` but cannot specify any of the access parameters (`model_access_name`, `backend_roles`, or `add_backend_roles`). An exception will be thrown if the request has any access control related fields in the input: `You cannot specify model access control parameters because Security plugin or model access control is disabled on your cluster.`
+
+
+
+
+## Updating a model group
+
+Updating a model group request is very similar to register model group request. Additionally, user needs to specify the `model_group_id` field in the request to which he/she needs to make an update.
+
+
+### Path and HTTP method
+
+```
+PUT /_plugins/_ml/model_groups/<model_group_id>_update
+```
+
+* A user can make updates to a model group to which he/she has access which is determined by the access mode of the model group.
+* In a model_access_control enabled cluster, the model group owner or an admin user can update all fields. Whereas, any other user who shares one or more backend roles with the model group can update the `name` and `description` fields only.
+* In a security/model_access_control disabled cluster, any user can make updates to any model group but cannot specify any of the access fields.
+* Consider a model group created by User1 with IT backend_role set to it. Since user1 is the owner, he/she will be able to update all the fields of the model group.
```

Comment:
Seems line 188 already covers line 190, 191
Line 187 covers line 192?

Line 190, 191, 192 are describing an example. Remove these lines or move these lines out of bulleted list?